frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -44,45 +52,166 @@ export async function action({ context, request }: Route.ActionArgs) {
+      }
+
+      context.session.inPersonSINCase = {
+        ...(context.session.inPersonSINCase ?? {}),
+        ...input,
+        primaryDocStep: formData.get('step') as string,
+      };
 
-  throw i18nRedirect('routes/protected/request.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/secondary-docs.tsx"
+      throw i18nRedirect('routes/protected/request.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/secondary-docs.tsx"
+    }
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`);
+    }
+  }
 }
 
 export default function PrimaryDocs({ loaderData, actionData, params }: Route.ComponentProps) {
   const { t } = useTranslation(handle.i18nNamespace);
 
   const fetcher = useFetcher<Info['actionData']>();
   const isSubmitting = fetcher.state !== 'idle';
-  const errors = fetcher.data?.errors;
+  const errors: { currentStatusInCanada?: string[]; documentType?: string[] } | undefined = fetcher.data?.errors ?? undefined;
+
+  const [step, setStep] = useState(1);
+
+  const [currentStatus, setCurrentStatus] = useState(loaderData.defaultFormValues.currentStatusInCanada);
+  const [secondaryInput, setSecondaryInput] = useState<JSX.Element | null>(null);
+
+  const handleCurrentStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    setCurrentStatus(event.target.value);
+  };
+
+  const handleNext = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (secondaryInput) {
+      setStep(step + 1);
+    }
+  };
+
+  useEffect(() => {
+    switch (currentStatus) {
+      case 'canadian-citizen-born-outside-canada':
+        setSecondaryInput(
+          <DocumentType defaultValue={loaderData.defaultFormValues.documentType} errorMessage={errors?.documentType?.[0]} />,
+        );
+        break;
+      case 'registered-indian-born-outside-canada':
+        setSecondaryInput(
+          <DocumentTypeRegisteredIndian
+            defaultValue={loaderData.defaultFormValues.documentType}
+            errorMessage={errors?.documentType?.[0]}
+          />,
+        );
+        break;
+      default:
+        setSecondaryInput(null); // Clear secondary input if no match
+    }
+  }, [currentStatus, errors, loaderData.defaultFormValues.documentType]);
+
+  return (
+    <>
+      <div className="flex justify-end">
+        <Button id="abandon-button" endIcon={faXmark} variant="link">
+          {t('protected:person-case.abandon-button')}
+        </Button>
+        <Button id="refer-button" endIcon={faExclamationCircle} variant="link">
+          {t('protected:person-case.refer-button')}
+        </Button>
+      </div>
+      <Progress className="mt-8" label="" value={30} />
+      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:primary-identity-document.page-title')}</PageTitle>
+
+      <fetcher.Form method="post" noValidate>
+        <input type="hidden" name="step" value={step.toString()} />
```

Comment:
I don't think a hidden field is necessary here if you are using `useState()` to handle rendering of the secondary input. You should just be able to call `setState('my-value')` and then render the second input by examining that value.

You may not even need to use `useState()` if you render conditionally based on the value of your `<CurrentStatusInCanada>` component.